wikiclass/feature_lists/wikidatawiki.py

```diff
@@ -14,16 +35,137 @@ class properties:
     DATE_OF_BIRTH = 'P569'
     DATE_OF_DEATH = 'P570'
 
-
 class items:
     """
     Mapping of english descriptions to item idenifiers
     """
     HUMAN = 'Q5'
 
+def _process_all_sources(item):
+
+	list_sources_in_JSON = []
+	
+	try:
+		for property in item.claims:
+			list_of_properties = property
+			for claim in item.claims[list_of_properties]:
+				list_of_claims = claim
+				for i, source in enumerate(list_of_claims.sources):
+					for index_list_1, index_list_2 in source.items(): 
+						sources_in_JSON = index_list_2[0].toJSON()
+						list_sources_in_JSON.append(sources_in_JSON['datavalue']['value'])
+			
+		return list_sources_in_JSON
+	except:
+		return 0 #if the revision does not contain any sources, return 0
+
+def _process_wikimedia_sources(list_all_sources):
+
+	wikimedia_sources = set()
+	result_list = []
+    
+	wdir_path = os.path.dirname(os.path.realpath(__file__)) #get the current working directory path
+	csv_path = os.path.join(wdir_path, 'excluded_qids.csv')
+	
+	#add content from the CSV to set
+	with open(csv_path) as csvfile:
```

Comment:
This opens the file every time a revision is scored.  Instead, read the file and put the contents into a global variable (as I'd suggested in a previous comment) and just reference that.  That way you only open the file once.